### PR TITLE
SFR-2154: Implement general error page designs

### PR DESCRIPTION
**Pull request by jackiequach**
[Jira Ticket](https://newyorkpubliclibrary.atlassian.net/browse/SFR-2154)
[Figma Designs](https://www.figma.com/design/K09KeGAVEXpleym34nTlue/DRB---Error-Pages?node-id=4302-6956&t=qpC6VsH5mJWaMFlv-1)

### This PR does the following:
- Updates the Feedback component to use context to be able to call the open function across the app
- Implements general error page designs

### Testing requirements & instructions: 
-

### CODE REVIEW

File: playwright/integration/landing.spec.ts [hunk 1]
```diff
@@ -28,6 +28,6 @@ test("Shows error boundary for invalid collection", async ({ page, port }) => {
   await page.goto(`http://localhost:${port}${INVALID_COLLECTION_PATH}`);
 
   const alert = page.getByRole("alert");
-  const errorText = alert.getByText("An error 500 occurred on server");
+  const errorText = alert.getByText("Something went wrong on our end");
```
Comment (kylevillegas93) [inline playwright/integration/landing.spec.ts:31]: Are there updated designs? 

In the draft design, I see "Oops! Something went wrong while loading this page."

File: src/components/Feedback/Feedback.tsx [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+    if (isError) setIsError(false);
+    if (notificationText) setNotificationText(null);
+    onClose();
+    setView("form");
+  };
 
   const handleFeedbackSubmit = (
     values: React.ComponentProps<typeof FeedbackBox>["onSubmit"]
   ) => {
     submitFeedback({
```
Comment (kylevillegas93) [inline src/components/Feedback/Feedback.tsx:40]: Just for my knowledge, where is feedback submitted?

File: src/components/Feedback/Feedback.tsx [hunk 3]
```diff
@@ -1,18 +1,47 @@
...
+  } = useContext(FeedbackContext);
+
+  useEffect(() => {
+    if (isError) setDescriptionText(ERROR_DESCRIPTION_TEXT);
+    else setDescriptionText(DEFAULT_DESCRIPTION_TEXT);
+  }, [isError]);
+
+  const onCloseAndReset = () => {
+    if (isError) setIsError(false);
+    if (notificationText) setNotificationText(null);
```
Comment (kylevillegas93) [inline src/components/Feedback/Feedback.tsx:32]: Could be simplify this and just setIsError(false) and setNotificationText(null) regardless of previous state? 

Or would that cause a unnecessary rerender?

File: src/pages/_error.tsx [hunk 4]
```diff
@@ -1,23 +1,98 @@
...
+    subText:
+      "The page you were looking for doesn't exist or may have moved elsewhere. ",
+    tryText: "Try a different URL or ",
+  },
+  400: {
+    overline: "error 500",
+    heading: "There was an unexpected error.",
+    subText: "We couldn't process your request at this time. ",
+    tryText: "Try refreshing the page or ",
+  },
```
Comment (kylevillegas93) [inline src/pages/_error.tsx:40]: Hmm 400 means that the request input was invalid. Should we reflect that accordingly in the text?

File: src/pages/_error.tsx [hunk 5]
```diff
@@ -1,23 +1,98 @@
...
+} from "@nypl/design-system-react-components";
+import { FeedbackContext } from "../context/FeedbackContext";
+
+const ERROR_PERSISTS = " if the error persists.";
+
+const getNotificationText = (statusCode) => {
+  return `You are asking for help or information about a page error (error code ${statusCode})`;
+};
+
+const errorMap = {
```
Comment (kylevillegas93) [inline src/pages/_error.tsx:21]: Should we also add a 403 mapping for limited access?

**Comment by kylevillegas93** [comment]
Overall, looks good! Just a few minor questions and comments. 

One thing that I have found helpful in the past when implementing the designs is to link to the Figma directly in the PR description. 

What is also good is to take screenshots of the different stages/pages. This is helpful for the reviewer to just help spot check between the designs and what is implemented.

File: playwright/integration/landing.spec.ts (hunk 1 above)
Comment (jackiequach) [reply playwright/integration/landing.spec.ts:31]: The updated designs are under V2 - FINAL

File: src/components/Feedback/Feedback.tsx (hunk 2 above)
Comment (jackiequach) [reply src/components/Feedback/Feedback.tsx:40]: There is an Airtable with the data. I believe Lyndsey and Dmitri have access.

File: src/pages/_error.tsx (hunk 5 above)
Comment (jackiequach) [reply src/pages/_error.tsx:21]: 403 is not returned for the limited access routes. For invalid auth, the user is redirected to the login page.

File: src/components/Feedback/Feedback.tsx (hunk 2 above)
Comment (kylevillegas93) [reply src/components/Feedback/Feedback.tsx:40]: Thanks!

**Review by kylevillegas93 — APPROVED** [review]
Nice work!

**Pull request merged**